scripts/plugins/analyze-go-test-from-bazel-output.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+# parse bazel go test case index log
+# param $1 file path
+function parse_bazel_go_test_index_log() {
+    indexReg="((===|---) (RUN|PASS|FAIL))|-- Test timed out at|FLAKY|(====================( Test output for //|=+))"
+    local logPath="$1"
+    grep --color -nE "$indexReg" "$logPath" >bazel-go-test-index.log
+}
+
+function parse_bazel_target_output_log() {
+    local logPath="$1"
+
+    local indexes=($(grep -nE "====================( Test output for //|=+)" ${logPath} | grep -oE "^\d+"))
+    local p=0
+    while [ $((2 * p)) -lt "${#indexes[@]}" ]; do
+        saveFlag="bazel-target-output-L${indexes[$((2 * p))]}-${indexes[$((2 * p + 1))]}"
+        sed -n "${indexes[$((2 * p))]},${indexes[$((2 * p + 1))]}p" "${logPath}" >"$saveFlag.log"
+        local append=""
+
+        if grep -E "^-- Test timed out at" "$saveFlag.log" >/dev/null; then
+            append="$append.timeout"
+        elif grep -E "^--- FAIL" "$saveFlag.log" >/dev/null; then
+            append="$append.fail"
+        elif cat "$saveFlag.log" | grep -E "(---|===) (PASS|FAIL|RUN)" | grep -oE "\bTest\w+\b" | sort | uniq -c | grep "^\s*1\b" >/dev/null; then
```

Comment:
Whether the `$saveFlag.log` will be unusually large, e.g. more than 1GB, `cat` may lead to container OOM.

Reply:
fixed, replace it with `grep pattern file`